pkg/template/core/starlark_value.go

```diff
@@ -107,34 +110,54 @@ func (e StarlarkValue) asInterface(val starlark.Value) interface{} {
 	}
 }
 
-func (e StarlarkValue) dictAsInterface(val *starlark.Dict) interface{} {
+func (e StarlarkValue) dictAsInterface(val *starlark.Dict) (interface{}, error) {
 	result := orderedmap.NewMap()
 	for _, item := range val.Items() {
 		if item.Len() != 2 {
 			panic("dict item is not KV")
 		}
-		result.Set(e.asInterface(item.Index(0)), e.asInterface(item.Index(1)))
+		key, err := e.asInterface(item.Index(0))
+		if err != nil {
+			return nil, err
+		}
+		value, err := e.asInterface(item.Index(1))
+		if err != nil {
+			return nil, err
+		}
+		result.Set(key, value)
 	}
-	return result
+	return result, nil
 }
 
-func (e StarlarkValue) structAsInterface(val *StarlarkStruct) interface{} {
+func (e StarlarkValue) structAsInterface(val *StarlarkStruct) (interface{}, error) {
 	// TODO accessing privates
 	result := orderedmap.NewMap()
+	var err error
 	val.data.Iterate(func(k, v interface{}) {
-		result.Set(k, e.asInterface(v.(starlark.Value)))
+		if err == nil {
```

Comment:
should we add IterateWithErr?

Reply:
No. That would just make too much sense.

Fixed.